### ISSUE TITLE
Create a culture-aware GherkinDialectProvider

### ISSUE THREAD

**Issue by dirkrombauts** [picklesdoc/pickles#349]
# The problem

Consider this date: 1/5/2016. Is that the first of May or the fifth of January 2016? That depends on who you ask. A British citizen will tell you it's the first of May, but an American citizen will tell you that it's January the fifth.

Cucumber cannot deal with these differences: you can specify the language of the feature file, but nothing more. You can write `# language: en` to indicate that your feature file is in English. There's no way of indicating whether you should interpret dates in British or American fashion.

SpecFlow does support these differences. For British you write `# language: en-GB` and for American you write `# langague: en-US`.

Currently, Pickles behaves like Cucumber. The goal of this story is to modify its behaviour so that Pickles supports the same range of languages as SpecFlow.

## The Solution

In [WhenParsingLocalizedFeatureFiles.cs](https://github.com/picklesdoc/pickles/blob/develop/src/Pickles/Pickles.Test/WhenParsingLocalizedFeatureFiles.cs), add a new test:

```cs
[Test]
public void WhenIndicatingTheLanguageAndCultureInTheFeature_ThenCanParseTheFeature()
{
     string featureText =
@"# language: nl-BE
Functionaliteit: Test de Cultuur

Scenario: Het Scenario
    Stel dat ik 50 ingeef
    En dat ik 70 ingeef
    Als ik plus druk
    Dan moet het resultaat 120 zijn";
 
     var configuration = this.Configuration;
     var parser = new FeatureParser(FileSystem, configuration);
     Feature feature = parser.Parse(new StringReader(featureText));

     Check.That(feature).IsNotNull();
}
```

In the [Pickles](https://github.com/picklesdoc/pickles/tree/develop/src/Pickles/Pickles) project, create a new class `CultureAwareDialectProvider` that inherits from `Gherkin.GherkinDialectProvider`. Override the `GetDialect` method with an implementation along these lines:

```
if there is a culture, then take the part before the `-` and call the base method with that.
if not, just call the base method
```

In the [FeatureParser](https://github.com/picklesdoc/pickles/blob/develop/src/Pickles/Pickles/FeatureParser.cs) class, use the new class on line 72:

```cs
new Gherkin.TokenMatcher(new CultureAwareDialectProvider(language)));
```

**Comment by dirkrombauts** [comment]
This issue would resolve #308

**Comment by dirkrombauts** [comment]
Released in version [2.10.0](https://github.com/picklesdoc/pickles/releases/tag/v2.10.0).
